### ISSUE TITLE
Method 'Microsoft.Identity.Client.PublicClientApplication..ctor' not found.

### ISSUE THREAD

**Issue by RhomGit** [AzureAD/microsoft-authentication-library-for-dotnet#539]
Microsoft.Identity.Client 1.1.2-preview0008

When using the PublicClientApplication constructor "string, string, TokenCache" from my Xamarin/Android app the function bombs when it tries to enter the function that calls the constructor with the error:
_"Method 'Microsoft.Identity.Client.PublicClientApplication..ctor' not found."_

Also, when I try and use the TokenCacheHelper from https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/token-cache-serialization 
in my Xamarin/Android app like this:
TokenCache tc = TokenCacheHelper.GetUserCache();
I get the error:
_"Could not resolve type with token 0100006d (from typeref, class/assembly Microsoft.Identity.Client.TokenCacheExtensions, Microsoft.Identity.Client, Version=1.1.2.0, Culture=neutral, PublicKeyToken=0a613f4dd989e8ae)"_

Both of these calls seems to work fine from my WPF app.

**Comment by jennyf19** [comment]
@RhomGit -
The PublicClientApplication constructor "string, string, TokenCache" is only available for Desktop and NetCore apps. 
For Xamarin/Android, you will need to use one of these [constructors](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/dev/src/Microsoft.Identity.Client/PublicClientApplication.cs#L48)
You can see how we use it in our [Xamarin Sample App](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/dev/tests/dev%20apps/XForms/XForms/App.xaml.cs#L79)

**Comment by RhomGit** [comment]
Yep, I figured that out through trial and error. I lost a bit of time with NuGet thinking it was incompatible libraries but it is all working fine now. Thanks for your sample, I noticed you handle the RedirectUri differently to the way that I was so I tried switching to your method but it didn't seem to work. I have reverted to the method described in section 6b) here:
https://github.com/Azure-Samples/active-directory-b2c-xamarin-native
